packages/arb-avm-cpp/CMakeLists.txt

```diff
@@ -57,7 +57,25 @@ set(AVM_VERSION_STRING ${AVM_VERSION_MAJOR}.${AVM_VERSION_MINOR}.${AVM_VERSION_P
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
 
 option(ENABLE_JEMALLOC "Enable jemalloc" OFF)
-option(AVM_BUILD_TESTING "Build tests" ON)
+
+set(AVM_FUZZER "" CACHE STRING "Build fuzzing target for the requested fuzzer")
+
+if (${AVM_FUZZER} MATCHES "afl")
+    SET(CMAKE_C_COMPILER "/usr/local/bin/afl-clang-lto" CACHE PATH "" FORCE)
+    SET(CMAKE_CXX_COMPILER "/usr/local/bin/afl-clang-lto++" CACHE PATH "" FORCE)
+    add_compile_options(-fsanitize=address)
+    add_link_options(-fsanitize=address)
+elseif (${AVM_FUZZER} MATCHES "libfuzzer")
+    SET(CMAKE_C_COMPILER "/usr/bin/clang" CACHE PATH "" FORCE)
+    SET(CMAKE_CXX_COMPILER "/usr/bin/clang++" CACHE PATH "" FORCE)
+    add_compile_options(-fsanitize=address,fuzzer-no-link)
+    add_link_options(-fsanitize=address,fuzzer)
+elseif (${AVM_FUZZER} MATCHES "honggfuzz")
+    SET(CMAKE_C_COMPILER "/usr/local/bin/hfuzz-clang" CACHE PATH "" FORCE)
+    SET(CMAKE_CXX_COMPILER "/usr/local/bin/hfuzz-clang++" CACHE PATH "" FORCE)
+    add_compile_options(-fsanitize=address)
+    add_link_options(-fsanitize=address)
+endif ()
```

Comment:
I don't love the hardcoded paths. Maybe better to move to setting via the command line instead via `cmake -DCMAKE_CXX_COMPILER=afl-clang-lto++ ..` instead? I was looking at https://github.com/jefftrull/json_spirit/blob/develop/fuzzing/CMakeLists.txt for example

Reply:
My goal here was for this to be as easy to setup as possible. Ideally I'd specify a non-absolute path in the CMakeLists.txt but but that doesn't seem possible.